src/Client.php

```diff
@@ -85,15 +101,20 @@ class Client implements ServerClient
      *
      * @param  string $server      Full address of the XML-RPC service
      *                             (e.g. http://time.xmlrpc.com/RPC2)
-     * @param  \Laminas\Http\Client $httpClient HTTP Client to use for requests
+     * @param  HttpClientInterface|null $httpClient HTTP Client to use for requests
      */
-    public function __construct($server, ?Http\Client $httpClient = null)
-    {
+    public function __construct(
+        string $server,
+        HttpClientInterface|null $httpClient = null,
+        RequestFactoryInterface|null $requestFactory = null,
+        StreamFactoryInterface|null $streamFactory = null
```

Comment:
None of these properties should be nullable - they are all hard dependencies that this class _requires_ in order to operate.


```suggestion
        private readonly HttpClientInterface $httpClient,
        private readonly RequestFactoryInterface $requestFactory,
        private readonly StreamFactoryInterface $streamFactory,
```